samples/pmic/native/npm2100_one_button/src/main.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/logging/log_ctrl.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/mfd/npm2100.h>
+#include <zephyr/drivers/regulator.h>
+
+#define FAST_FLASH_MS	100
+#define SLOW_FLASH_MS	500
+#define PRESS_SHORT_MS	1000
+#define PRESS_MEDIUM_MS 5000
+
+static int flash_time_ms = SLOW_FLASH_MS;
+static bool shipmode;
+
+static const struct gpio_dt_spec led = GPIO_DT_SPEC_GET(DT_ALIAS(led0), gpios);
+static const struct device *pmic = DEVICE_DT_GET(DT_NODELABEL(npm2100_pmic));
+static const struct device *regulators = DEVICE_DT_GET(DT_NODELABEL(npm2100_regulators));
+
+static void event_callback(const struct device *dev, struct gpio_callback *cb, uint32_t events)
+{
+	static int press_t;
+
+	if (events & BIT(NPM2100_EVENT_SYS_SHIPHOLD_FALL)) {
+		press_t = k_uptime_get();
+	}
+
+	if (events & BIT(NPM2100_EVENT_SYS_SHIPHOLD_RISE)) {
+		press_t = k_uptime_get() - press_t;
```

Comment:
I think it would be more readable if you use another variable to hold the time delta value.